src/fhirdefs/FHIRDefinitions.ts

```diff
@@ -43,97 +31,27 @@ export class FHIRDefinitions implements Fishable {
-    return cloneJsonMapValues(this.logicals);
-  }
-
-  allProfiles(): any[] {
-    return cloneJsonMapValues(this.profiles);
-  }
-
-  allExtensions(): any[] {
-    return cloneJsonMapValues(this.extensions);
-  }
-
-  allTypes(): any[] {
-    return cloneJsonMapValues(this.types);
-  }
-
-  allValueSets(): any[] {
-    return cloneJsonMapValues(this.valueSets);
-  }
-
-  allCodeSystems(): any[] {
-    return cloneJsonMapValues(this.codeSystems);
-  }
-
-  allImplementationGuides(): any[] {
-    return cloneJsonMapValues(this.implementationGuides);
-  }
-
   allPredefinedResources(): any[] {
     return cloneJsonMapValues(this.predefinedResources);
   }
 
   add(definition: any): void {
     // For supplemental FHIR versions, we only care about resources and types,
     // but for normal packages, we care about everything.
-    if (definition.resourceType === 'StructureDefinition') {
-      if (
-        definition.type === 'Extension' &&
-        definition.baseDefinition !== 'http://hl7.org/fhir/StructureDefinition/Element' &&
-        !this.isSupplementalFHIRDefinitions
-      ) {
-        addDefinitionToMap(definition, this.extensions);
-      } else if (
-        definition.kind === 'primitive-type' ||
-        definition.kind === 'complex-type' ||
-        definition.kind === 'datatype'
-      ) {
-        addDefinitionToMap(definition, this.types);
-      } else if (definition.kind === 'resource') {
-        if (definition.derivation === 'constraint') {
-          if (!this.isSupplementalFHIRDefinitions) {
-            addDefinitionToMap(definition, this.profiles);
-          }
-        } else {
+    if (this.isSupplementalFHIRDefinitions) {
+      if (definition.resourceType === 'StructureDefinition') {
+        if (
+          definition.kind === 'primitive-type' ||
+          definition.kind === 'complex-type' ||
+          definition.kind === 'datatype'
+        ) {
+          addDefinitionToMap(definition, this.types);
+        } else if (definition.kind === 'resource' && definition.derivation !== 'constraint') {
           addDefinitionToMap(definition, this.resources);
         }
-      } else if (definition.kind === 'logical') {
-        if (definition.derivation === 'specialization') {
-          addDefinitionToMap(definition, this.logicals);
-        } else if (!this.isSupplementalFHIRDefinitions) {
-          addDefinitionToMap(definition, this.profiles);
-        }
       }
-    } else if (definition.resourceType === 'ValueSet' && !this.isSupplementalFHIRDefinitions) {
-      addDefinitionToMap(definition, this.valueSets);
-    } else if (definition.resourceType === 'CodeSystem' && !this.isSupplementalFHIRDefinitions) {
-      addDefinitionToMap(definition, this.codeSystems);
-    } else if (
-      definition.resourceType === 'ImplementationGuide' &&
-      !this.isSupplementalFHIRDefinitions
-    ) {
-      addDefinitionToMap(definition, this.implementationGuides);
+    } else {
+      super.add(definition);
     }
   }
```

Comment:
As noted above, I think that `addDefinitionToMap` shouldn't be exported from FPL (because it is an implementation detail).  Since the FPL `FHIRDefinition` adds the appropriate types to the appropriate maps already, then we can get away with using `super.add` instead (since the difference in this code is just _when_ to add, not _how_ to add).  I tested this implementation instead and it seemed to work (or at least pass the unit tests):
```ts
  add(definition: any): void {
    // For supplemental FHIR versions, we only care about resources and types,
    // but for normal packages, we care about everything.
    if (this.isSupplementalFHIRDefinitions) {
      if (
        definition.resourceType === 'StructureDefinition' &&
        (definition.kind === 'primitive-type' ||
          definition.kind === 'complex-type' ||
          definition.kind === 'datatype' ||
          (definition.kind === 'resource' && definition.derivation !== 'constraint'))
      ) {
        super.add(definition);
      }
    } else {
      super.add(definition);
    }
  }
```

Reply:
Simplified the `if` logic and swapped out `addDefinitionToMap` with `super.add` in [1e1e8a9](https://github.com/FHIR/sushi/pull/1026/commits/1e1e8a9c1da76bf964a68f525572a28c3c562363).

I also confirmed that the definition that gets passed in to `super.add` will still fall into the same cases in FPL, so `addDefinitionToMap` still gets called with the correct map from FPL, so this implementation seems correct to my eye and the tests.